### DISCUSSION TITLE
Fabric shim compatibility

### DISCUSSION THREAD

**Discussion by marchinm** [hyperledger/fabric-chaincode-java#245] [Q&A]
Folks,

Does anyone know if fabric-chaincode-java version 1.4.4 is compatible with Fabric version 2.4.1?

I'm trying to set up a chaincode container and it's failing to connect to the peer, indicating "Network closed for unknown reason"

Here's the log from the failed pod:
[chaordicledger-org1-peer1-cc-lightswitch-8b647d977-46lvs-1654896351819580806.log](https://github.com/hyperledger/fabric-chaincode-java/files/8882395/chaordicledger-org1-peer1-cc-lightswitch-8b647d977-46lvs-1654896351819580806.log)

I should node: I have a GO-based chaincode container that successfully connects to the peer and seems to work flawlessly.

Please advise, thanks!

**Comment by mbwhite** [comment]
Hello;  'wire-protocol' wise - yes they are completely compatible - so if the Java contract/chaincode can be started ok in the JVM, then it will connect to the peer.  

If the peer is launching the chaincode it will load the JavaEnv docker image to run the code. But this image can be different between Fabric 1.4/2.4.   From memory though the JavaEnv should be very similar. 

It's worth also checking what additional libraries your chaincode uses. For example I saw a chaincode last week that also included the Quarkus libraries.  That completely failed to start networking as a result. 

*however* Any Fabric 1.4 component is now out of Long Term Support; and unless absolutely critical we won't be updating any 1.4 libraries any more.  So would strongly recommend updating to the 2.4 fabric-chaincode-java. 

API wise you shouldn't see any issues migrating
